packages/flutter/lib/src/widgets/form.dart

```diff
@@ -293,19 +293,45 @@ class FormState extends State<Form> {
   /// returns true if there are no errors.
   ///
   /// The form will rebuild to report the results.
+  /// See also: [validateGranually]
   bool validate() {
     _hasInteractedByUser = true;
     _forceRebuild();
-    return _validate();
+    return _validate().isValid;
   }
 
-  bool _validate() {
+
+  /// Validates every [FormField]s that is a descendant of this [Form], and
+  /// returns a [Map] which keys' are [Key]s of the descendant [FormField]s
+  /// and values' are their corresponding validation results.
+  ///
+  /// Any [FormField] that has not been asigned a key will be ignored in the result.
+  /// To get the absolute validation value of the Form, Consider using [validate]
+  ///
+  /// Common usage of this method is when you need draw the user's attention
+  /// to the invalid field(s) using their widget key(s).
+  ///
+  /// The form will rebuild to report the results.
+  Map<Key, bool> validateGranually() {
+    _hasInteractedByUser = true;
+    _forceRebuild();
+    return _validate().fieldsValidationStatus;
+  }
+
+  _FormValidationStatus _validate() {
+    final Map<Key, bool> fieldsValidationStatus = <Key, bool>{};
     bool hasError = false;
     String errorMessage = '';
     for (final FormFieldState<dynamic> field in _fields) {
       hasError = !field.validate() || hasError;
       errorMessage += field.errorText ?? '';
+      final Key? key = field.widget.key;
+      if (key != null) {
+        fieldsValidationStatus[key] = field.validate();
```

Comment:
`validate()` is called twice in this method.

Reply:
Also I think we probably shouldn't change `_validate`'s implementation like this. `validate` only returns a bool and the `Map` populated ends up getting thrown away.

Reply:
> validate() is called twice in this method.

You're right 👍 I'll fix.

> the Map populated ends up getting thrown away.

The map is used in the new method `validateGranulay`. 
`validate` and `validateGranulay` should share the same logic except for the returned value. That's why I added the wrapper class `_FormValidationStatus` and avoided duplicating code, for example, the code for announcing the error.
I could extract the shared code to a private method and call it in `_validate` and `validateGranualy`, but any future changes that involve performing some side effects on validation will also need to be written twice.
It's just an opinion, please lmk if you disagree I'll make the changes by extraction.

Reply:
@SharbelOkzan Would it work if you passed a optional map to the `_validate` method?

So it would be
```dart
  Map<Key, bool> validateGranually() {
    final Map<Key, bool> results = <Key, bool>{};
    _hasInteractedByUser = true;
    _forceRebuild();
    _validate(results);
    return results;
  }

  bool validate() {
    _hasInteractedByUser = true;
    _forceRebuild();
    return _validate();
  }

  bool _validate([Map<Key, bool>? optionalResults]) { // This can also be _validate({}) for named params.
    bool hasError = false;
    String errorMessage = '';
    for (final FormFieldState<dynamic> field in _fields) {
      hasError = !field.validate() || hasError;
      errorMessage += field.errorText ?? '';
      final Key? key = field.widget.key;
      if (optionalResults != null && key != null) {
        optionalResults[key] = field.validate();
      }
    }

    if (errorMessage.isNotEmpty) {
      final TextDirection directionality = Directionality.of(context);
      if (defaultTargetPlatform == TargetPlatform.iOS) {
        unawaited(Future<void>(() async {
          await Future<void>.delayed(_kIOSAnnouncementDelayDuration);
          SemanticsService.announce(errorMessage, directionality, assertiveness: Assertiveness.assertive);
        }));
      } else {
        SemanticsService.announce(errorMessage, directionality, assertiveness: Assertiveness.assertive);
      }
    }
    return !hasError;
  }
```